File: frontend/src/types/Project.ts
export type ProjectStatus = "ACTIVE" | "COMPLETED" | "ON_HOLD" | "CANCELLED";

export interface Project {
    projectId: string;
    projectName: string;
    departmentId: string;
    departmentName: string;
    ownerId: string;
    ownerName: string;
    status: ProjectStatus;
    startDate: string;
    endDate: string | null;
}

File: backend/src/main/java/com/malini/enfos/backend/model/Project.java
package com.malini.enfos.backend.model;

import com.malini.enfos.backend.enums.ProjectStatus;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Project {
    private String projectId;
    private String projectName;
    private String departmentId;
    private String ownerId;
    private ProjectStatus status;
    private String startDate;
    private String endDate;
}
